app/models/tigerdata_schema.rb

```diff
@@ -0,0 +1,53 @@
+# frozen_string_literal: true
+
+# Defines the Tigerdata schema in MediaFlux along with the
+# document types allowed. At this point we only support "projects"
+class TigerdataSchema
+  attr_accessor :schema_name, :schema_description
+
+  def initialize(schema_name: nil, schema_description: nil, session_id:)
+    @schema_name = schema_name || "tigerdata"
+    @schema_description = schema_description || "TigerData Metadata"
+    @session_id = session_id
+  end
+
+  def create
+    define_schema_namespace
+    define_project
+  end
+
+  def define_schema_namespace
+    schema_request = Mediaflux::Http::SchemaCreateRequest.new(name: @schema_name,
+                                                              description: @schema_description,
+                                                              session_token: @session_id)
+    schema_request.resolve
+  end
+
+  def define_project
+    fields_request = Mediaflux::Http::SchemaFieldsCreateRequest.new(
+      schema_name: @schema_name,
+      document: "project",
+      description: "Project Metadata",
+      fields: project_schema_fields,
+      session_token: @session_id
+    )
+    fields_request.resolve
+  end
+
+  def project_schema_fields
+    # TODO: I think these fields should be their own Ruby class and we can make them
```

Comment:
@hectorcorrea is this todo still relevant?

Reply:
I think so. I'll leave it for now and if by the time we are done messing up with the schema we don't need it we can remove it.